test/parallel/test-http2-https-fallback.js

```diff
@@ -115,12 +115,15 @@ function onSession(session) {
     common.mustCall(onRequest)
   );
 
-  server.on('unknownProtocol', common.mustCall((socket) => {
+  server.once('unknownProtocol', common.mustCall((socket) => {
     socket.destroy();
-  }, 2));
+  }));
 
   server.listen(0);
 
+  // Test for checking undefined timeout (code: https://git.io/vdzE9)
+  server.setTimeout(undefined);
```

Comment:
This feels out of place in this test. I would suggest removing it. If it needs to be tested, there must be a more appropriate existing test to insert it in (one that already deals with timeouts). (The link to git.io is also unnecessary.)

Reply:
Removing the test or removing the check for `this.timeout` in `connectionListener`?
I think latter should be done as timeout is always expected to be a number (either default one or the one explicitly set).

Reply:
Removing the test for `setTimeout(undefined)`. It doesn't relate to http2 https fallback and is likely to just confuse people in the future.

Reply:
Removed the test for `setTimeout(undefined)` and updated the commit